larky/src/main/resources/vendor/Crypto/Cipher/_mode_cfb.star

```diff
@@ -212,34 +145,71 @@ def CfbMode(block_cipher, iv, segment_size):
 
         if self.decrypt not in self._next:
-            fail(" TypeError(\"decrypt() cannot be called after encrypt()\")")
+            fail("TypeError: decrypt() cannot be called after encrypt()")
         self._next = [ self.decrypt ]
 
         if output == None:
-            plaintext = create_string_buffer(len(ciphertext))
+            plaintext = bytearray()
         else:
             plaintext = output
 
-            if not is_writeable_buffer(output):
+            if not types.is_bytearray(output):
                 fail(" TypeError(\"output must be a bytearray or a writeable memoryview\")")
 
             if len(ciphertext) != len(output):
-                fail(" ValueError(\"output must have the same length as the input\"\n                                 \"  (%d bytes)\" % len(plaintext))")
+                fail("ValueError: output must have the same length as the input (%d bytes)" % len(ciphertext))
+
+        result = self._state.decrypt(ciphertext, plaintext)
 
-        result = raw_cfb_lib.CFB_decrypt(self._state.get(),
-                                         c_uint8_ptr(ciphertext),
-                                         c_uint8_ptr(plaintext),
-                                         c_size_t(len(ciphertext)))
         if result:
-            fail(" ValueError(\"Error %d while decrypting in CFB mode\" % result)")
+            fail("ValueError: Error %d while decrypting in CFB mode" % result)
 
-        if output == None:
-            return get_raw_buffer(plaintext)
-        else:
-            return None
+        if output != None:
+            return
+
+        return plaintext
     self.decrypt = decrypt
+
+    def __init__(self, block_cipher, iv, segment_size):
+        """Create a new block cipher, configured in CFB mode.
+
+        :Parameters:
+          block_cipher : C pointer
+            A smart pointer to the low-level block cipher instance.
+
+          iv : bytes/bytearray/memoryview
+            The initialization vector to use for encryption or decryption.
+            It is as long as the cipher block.
+
+            **The IV must be unpredictable**. Ideally it is picked randomly.
+
+            Reusing the *IV* for encryptions performed with the same key
+            compromises confidentiality.
+
+          segment_size : integer
+            The number of bytes the plaintext and ciphertext are segmented in.
+        """
+
+        self._state = _JCrypto.Cipher.CFBMode(block_cipher, iv, segment_size*8)
+
+        self.block_size = len(iv)
+        """The block size of the underlying cipher, in bytes."""
+
+        self.iv = _copy_bytes(None, None, iv)
+        """The Initialization Vector originally used to create the object.
+        The value does not change."""
+
+        self.IV = self.iv
+        """Alias for `iv`"""
+
+        self._next = [ self.encrypt, self.decrypt ]
+        return self
+
+    self = __init__(self, block_cipher, iv, segment_size)
     return self
 
+def divmod(x, y):
```

Comment:
We have a built in divmod as well. This is not necessary.

Reply:
removed